.github/workflows/docker.yml

```diff
@@ -0,0 +1,67 @@
+# The present workflow was made based on the following references:
+# - https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching
+# - https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md
+---
+name: Docker
+
+on: push
+
+concurrency:
+  group: docker-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/setup-buildx-action@v1
+      - uses: actions/cache@v3
+        with:
+          key: v1-${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: v1-${{ runner.os }}-buildx-
+          path: /tmp/.buildx-cache
+      - uses: docker/login-action@v1 
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: ${{ env.REGISTRY }}
+      - uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+        id: meta
+      - name: Building image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
+          load: true
+      - name: Smoke test
+        run: docker run ${{ env.IMAGE_NAME }} --version
+      - name: Pushing image
+        if: contains(github.ref, 'refs/tags/v')
```

Comment:
@sestrella Oh, I see our plan is to build on every commit push and push the image when the ref is a tag. That makes sense. Although, I think we should also avoid the step `docker/login-action@v1` on every push. I guess the builds will fail when users out of the organization do a contribution since they're not part of our GH teams.

Reply:
@CristhianMotoche not really, since pushing tags requires write privileges to the repo, therefore, it can’t only be done by admins and maintainers, I don't think this is going to be an issue for external contributors.

Reply:
Good catch, about `docker/login-action@v1` I'm going to update that

Reply:
Hey @sestrella I think my previous comment was a bit confusing. I was referring to push commits from external contributors since I see that the `docker/login-action@v1` step is executed on every commit push. Then, it could fail for external contributors, couldn't it?